lib/Doctrine/Migrations/Configuration/Loader/AbstractFileLoader.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Configuration\Loader;
+
+use function dirname;
+use function realpath;
+
+/**
+ * @internal
+ */
+abstract class AbstractFileLoader implements Loader
+{
+    /**
+     * @param array<string,string> $input
+     *
+     * @return array<string,string>
+     */
+    protected function getDirectoryRelativeToFile(string $file, array $input) : array
```

Comment:
Should this method be `final`?

Also see https://github.com/localheinz/phpstan-rules#methodsfinalinabstractclassrule.